api/pkg/transformer/types/table/table.go

```diff
@@ -40,6 +52,162 @@ func NewRaw(columnValues map[string]interface{}) (*Table, error) {
+	var r io.ReadSeeker
+
+	if strings.HasPrefix(filePath, gcsPrefix) {
+		// Global file (GCS)
+		ctx := context.Background()
+		data, err := gsutil.ReadFile(ctx, filePath)
+
+		if err != nil {
+			return nil, err
+		}
+
+		r = bytes.NewReader(data)
+	} else {
+		// Local file
+		file, err := os.Open(filePath)
+		if err != nil {
+			return nil, err
+		}
+		r = file
+		defer file.Close()
+	}
+
+	fr, err := goparquet.NewFileReader(r)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get header
+	schema := fr.GetSchemaDefinition()
+	header := make([]string, 0)
+	for _, schemaCol := range schema.RootColumn.Children {
+		header = append(header, schemaCol.SchemaElement.GetName())
+	}
+
+	records := make([][]string, 0)
+	records = append(records, header) // column name in first row
+
+	// map col name to col number
+	colMap := make(map[string]int)
+	for i, colName := range header {
+		colMap[colName] = i
+	}
+
+	// Build data
+	for i := 0; i < int(fr.NumRows()); i++ {
+		row, err := fr.NextRow()
+		if err != nil {
+			return nil, err
+		}
+
+		newRow := make([]string, len(header))
+		for k, v := range row {
+			if vv, ok := v.([]byte); ok {
+				v = string(vv)
+			}
+
+			newRow[colMap[k]] = fmt.Sprint(v)
+		}
+		records = append(records, newRow)
+	}
+
+	return records, nil
+}
+
+// RecordsFromCsv reads a csv file which may be local (uploaded together with model) or global (from gcs).
+// The root directory for local will be same as where the model is stored (artifacts folder).
+// Data will be returned as [][]string where row 0 is the header containing all the column names.
+func RecordsFromCsv(filePath string) ([][]string, error) {
+	var r io.Reader
+
+	if strings.HasPrefix(filePath, gcsPrefix) {
+		// Global file (GCS)
+		ctx := context.Background()
+		data, err := gsutil.ReadFile(ctx, filePath)
+
+		if err != nil {
+			return nil, err
+		}
+
+		r = bytes.NewReader(data)
+	} else {
+		// Local file
+		file, err := os.Open(filePath)
+		if err != nil {
+			return nil, err
+		}
+		r = file
+		defer file.Close()
+	}
+
+	csvReader := csv.NewReader(r)
+	records, err := csvReader.ReadAll()
+
+	if err != nil {
+		return nil, err
+	}
+
+	return records, nil
+}
+
+// NewFromRecords create a new table from records in the form [][]string
+// The first row of the records array shall contain all the column names.
+// This function will check that there are at least 2 rows in records (header and data).
+// It will also check that the number of columns and the column names matches the schema specified,
+// before creating a new table to be returned
+func NewFromRecords(records [][]string, schema []*spec.Schema) (*Table, error) {
+
+	// check table has at least 1 row of data
+	if len(records) <= 1 {
+		return nil, fmt.Errorf("no data found")
+	}
+	// check no. of columns same as defined in schema
+	header := make(map[string]bool)
+	for _, colName := range records[0] {
+		header[colName] = true
+	}
+
+	if len(header) != len(schema) {
```

Comment:
I think we shouldn't enforce this. We should make `schema` as optional input and it's only used to override our inferred type.